packages/grafana-ui/src/components/IconButton/IconButton.tsx

```diff
@@ -9,6 +9,8 @@ import { Tooltip } from '../Tooltip/Tooltip';
 import { TooltipPlacement } from '../Tooltip/PopoverController';
 import { getFocusStyles, getMouseFocusStyles } from '../../themes/mixins';
 
+export type IconButtonVariant = 'primary' | 'destructive' | undefined;
```

Comment:
having undefined as an option feels strange, any reasons for this? 

Wonder if we should use the same type as ButtonVariant, but default to secondary. Alternative is to use IconButtonVariant = 'default' | 'primary' | 'destructive'

Reply:
I wasn't overly sure about how to map from the ButtonVariants to the IconButtonVariants. I think defaulting to `secondary` would feel more consistent with ButtonVariants that introducing `default` to IconButton. Will update. Thanks for the advice @torkelo